core/src/main/scala/dagr/core/execsystem/GraphNode.scala

```diff
@@ -37,11 +39,10 @@ import scala.collection.mutable.ListBuffer
   *                         is used to track submission, start, and end dates for the parent node.  Typically parent nodes
   *                         are [[dagr.core.tasksystem.Pipeline]]s.
   */
-class GraphNode(val taskId: BigInt,
-                var task: Task,
+class GraphNode(var task: Task,
                 predecessorNodes: Traversable[GraphNode] = Nil,
                 var state: GraphNodeState.Value = GraphNodeState.PREDECESSORS_AND_UNEXPANDED,
-                val parent: Option[GraphNode] = None) {
+                val parent: Option[GraphNode] = None) extends BaseGraphNode {
 
   private val _predecessors = new ListBuffer[GraphNode]()
   private val _predecessorsStatic = new ListBuffer[GraphNode]()
```

Comment:
I'll add it back in when I need it, which I anticipate being when I need to have a copy of the original execution graph; see scaladoc for `originalPredecessors`.  The other option is to have `_predecessors` be a `Map[GraphNode,Boolean` storing `true/false` for a predecessor if it has been satisfied.  I suppose I can make the latter change at a later date.  Thoughts?